diploma-project/components/Navbar.jsx

```diff
@@ -1,16 +1,16 @@
 import React, {useEffect, useState} from "react";
 import {app_name, default_profile_picture} from "../public/constants";
 import {useRouter} from "next/router";
-import {auth, firestore} from "../firebase";
+import {auth} from "../firebase";
 import Link from "next/link";
 import logOut from "../services/logOut";
 import Image from "next/future/image";
-import {getUser, User} from "../services/user";
-import {collection, onSnapshot, query, where} from "firebase/firestore";
+import { useUserStore } from "../store/userStorage";
 
 const Navbar = () => {
+    const [isLogged, setIsLogged] = useState(false);
     const [error, setError] = useState(null);
-    const [user, setUser] = useState(null);
+    const { user: user } = useUserStore((state) => ({user: state.user}));
```

Comment:
I think that you cam omit one user from the declaration:
`const { user } = useUserStore((state) => ({user: state.user}))`